lib/errors/index.js

```diff
@@ -20,7 +20,20 @@ class Exception {
   }
 
   getErrorDetails(config) {
-    return errorHelper.extractErrorInformation(null, this.error, config)
+    const errorDetails = errorHelper.extractErrorInformation(null, this.error, config)
+    errorDetails.expected = this.isExpected(config, errorDetails)
+
+    return errorDetails
+  }
+
+  isExpected(config, {type, message}) {
+    if (!this._expected) {
+      this._expected =
+        errorHelper.isExpectedErrorClass(config, type) ||
+        errorHelper.isExpectedErrorMessage(config, type, message)
+    }
+
+    return this._expected
```

Comment:
Looks like this was moved from `_linkExceptionToSegment()` which appears to be the only function that calls `getErrorDetails()`. Nice refactor/consolidation.